src/metabase/public_settings.clj

```diff
@@ -341,7 +341,7 @@
   "Current password complexity requirements"
   :visibility :public
   :setter     :none
-  :getter     (constantly password/active-password-complexity))
+  :getter     (constantly (password/active-password-complexity)))
```

Comment:
`constantly` is a function that takes a value and returns a function that will always return that value regardless of arguments. For example

```clj
(def n 1)

(defn- double-n [] (* n 2))

(double-n) ; => 2
(with-redefs [n 2] (double-n)) => 4

(def double-n-constantly (constantly (double-n)))

(double-n-constantly) ; => 2
(with-redefs [n 2] (double-n-constantly)) => 2
```

`(constantly (password/active-password-complexity))` will evaluate `(password/active-password-complexity)` at the time it defines the function and any return the value it got during that call for the rest of all time. It won't pick up any changes to what that function returns.

What you actually want to do here is replace

```clj
(constantly (password/active-password-complexity))
```

with

```clj
password/active-password-complexity
```

`:getter` is just supposed to be a function of zero args that returns the value. `password/active-password-complexity` matches that definition

Reply:
Makes sense, thanks! Double checking that tests now pass locally then I'll be ready for a full review.